WebAPI/Controllers/UserController.cs

```diff
@@ -123,5 +123,55 @@ public IActionResult Renew([FromBody] RenewModel body)
             var authUser = new AuthenticatedUserModel(token);
             return Ok(authUser);
         }
+
+        [AllowAnonymous]
+        [HttpPost("resetPassword")]
+        public async Task<IActionResult> ResetPassword([FromBody] ResetPasswordModel body)
+        {
+            body.Email = body.Email.ToLower();
+            var user = await _userService.GetUserByEmailAsync(body.Email);
+            if (user == null)
+            {
+                _logger.LogInformation($"Invalid password reset attemp. User {body.Email} doesn't exist.");
+                return ErrorResponse($"Používateľ {body.Email} neexistuje.");
+            }
+
+            if (!user.EmailConfirmed)
+            {
+                _logger.LogInformation($"Invalid password reset attemp. User {body.Email} didn't confirm email address.");
+                return ErrorResponse($"Najskôr prosím potvrď svoju emailovú adresu.");
+            }
+
+            string token = await _userService.GeneratePasswordResetTokenAsync(user);
+            token = Uri.EscapeDataString(token);
```

Comment:
EscapeDataString is called everytime after GenerateToken, can be part of function above?

Reply:
done